multicore_runtime/dispatcher.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2015 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""A WSGI app that, once configured, dispatches requests to user apps."""
+
+import httplib
+import logging
+import re
+
+from werkzeug.wrappers import Request
```

Comment:
Recommendation here seems to be:

```python
  from werkzeug import wrappers
```

And then using `wrappers.Request` and `wrappers.Response`. If you think `wrappers` is too opaque, you can use something like `as werkzeug_wrappers` or `as wzw` etc. to make it more usable.

The main argument towards only importing modules (and not individual elements) is if you need to use mocks or stubs. If you're always accessing it through `werkzeug.wrappers` you only need to mock it there, otherwise you'd have to mock it on **every** namesapce that is importing them, which would be basically impossible to do...

It's possible that there's no need to mock these particular objects... but the general advice is safe, so it's probably good to stick to it if possible.